### ISSUE TITLE
RCS1261: "Resource can be disposed asynchronously" does not take lock statement into account

### ISSUE THREAD

**Issue by NoahStolk** [dotnet/roslynator#1372]
**Product and Version Used**: 

Roslynator.Analyzers 4.9.0

**Steps to Reproduce**:

Consider the following code:

```cs
object fileStreamLock = new();

await SomeAsyncMethod();

async Task SomeAsyncMethod()
{
	lock (fileStreamLock)
	{
		using FileStream fs = new("test.txt", FileMode.OpenOrCreate);
		// Do something...
	}

	await Task.Yield(); // For demonstration purposes
}
```

`RCS1261` is reported, saying that the `FileStream` can be disposed asynchronously, but doing so results in [compiler error CS1996](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1996) (Cannot 'await' in the body of a 'lock' statement). This is not valid C#:

```cs
lock (fileStreamLock)
{
        await using FileStream fs = new("test.txt", FileMode.OpenOrCreate);
}
```
